File: frontend/src/domain/LearningSession.ts
// Keep in sync with backend/VocabuAI.Domain/LearningSession.cs
import type { LearningTask } from "./LearningTask";

export type LearningSession = {
  guid: string;
  userId: number;
  createdAt: string;
  tasks: LearningTask[];
};

File: backend/VocabuAI.Domain/LearningSession.cs
namespace VocabuAI.Domain.Learning;

public sealed class LearningSession
{
    public Guid Guid { get; init; }
    public int UserId { get; init; }
    public DateTimeOffset CreatedAt { get; init; }
    public IReadOnlyList<LearningTask> Tasks { get; init; } = Array.Empty<LearningTask>();
}
